### PR TITLE
docs: Add upgrade edp 3.8 guide

### PR DESCRIPTION
**Pull request by Rolika4**
# Pull Request Template

## Description
Please include a summary of the change and why it is needed.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [* ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

## Checklist:
- [* ] I have performed a self-review of my own code
- [* ] I have commented my code, particularly in hard-to-understand areas
- [ *] I have made corresponding changes to the documentation
- [ *] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ *] New and existing unit tests pass locally with my changes
- [ *] Pull Request contain one commit. I squash my commits

### CODE REVIEW

**Review by SergK — CHANGES_REQUESTED** [review]
@NikolayMarusenko please review as well

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    ```text
+    Old value: https://edp-headlamp-edp.<dns_wildcard>/*
+
+    New value: https://portal-edp.<dns_wildcard>/*
+    ```
+
+    !![Keycloak client](../assets/operator-guide/portal-keycloak.png "Keycloak client")
+
+4. Update webhooks for Github and GitLab
```
[the remark this answers is not in the document]
Comment (Rolika4) [reply docs/operator-guide/upgrade-edp-3.8.md]: add use-case for command-line update

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# Upgrade EDP v3.7 to 3.8
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:1]: Lets use KubeRocketCI instead of EDP

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    kubectl -n edp delete ingress edp-tekton-dashboard
+    kubectl -n edp delete ingress el-github-listener
+    kubectl -n edp delete EDPComponents sonar
+    kubectl -n edp delete EDPComponents nexus
+    kubectl -n edp delete EDPComponents argocd
+    kubectl -n edp delete EDPComponents main-keycloak
+    kubectl -n edp delete EDPComponents tekton
+    kubectl -n edp delete EDPComponents github
+    ```
+
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:22]: please add description that we remove CustomResourceDefinition edpcomponents.v1.edp.epam.com support and if user use CR in this CRD, before remove CRD user need to remove CR.

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+This section provides detailed instructions for upgrading the EPAM Delivery Platform to version 3.8.1. Follow the steps and requirements outlined below:
+
+
+1. Delete the following resources:
+
+    ```bash
+    kubectl -n edp delete ingress edp-headlamp
+    kubectl -n edp delete ingress edp-tekton-dashboard
+    kubectl -n edp delete ingress el-github-listener
+    kubectl -n edp delete EDPComponents sonar
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:15]: maybe we can remove all CR for EDPComponents CRD?

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+    ```text
+    Old value: https://edp-headlamp-edp.<dns_wildcard>/*
+
+    New value: https://portal-edp.<dns_wildcard>/*
+    ```
+
+    !![Keycloak client](../assets/operator-guide/portal-keycloak.png "Keycloak client")
+
+4. Remove all ArgoCD applications:
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:41]: Pleas add description why user need to remove ArgoCD applications,

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+            enabled: true
+            resources:
+            requests:
+                memory: "64Mi"
+                cpu: "50m"
+            limits:
+                memory: "128Mi"
+                cpu: "500m"
+            ingress:
+            enabled: true
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:87]: please fix spaces: https://github.com/epam/edp-tekton/blob/master/charts/pipelines-library/values.yaml#L364

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    Navigate to Keycloak realm with `Portal` client. Update `Valid redirect URIs` field:
+
+    ```text
+    Old value: https://edp-headlamp-edp.<dns_wildcard>/*
+
+    New value: https://portal-edp.<dns_wildcard>/*
+    ```
+
+    !![Keycloak client](../assets/operator-guide/portal-keycloak.png "Keycloak client")
+
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:40]: Please add description that this KubeRocketCI version require ArgoCD version with applicationset support and provide link to our guide with argocd RBAC hack.

**Review by Rolika4 — COMMENTED** [review]
done

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
+# Upgrade KubeRocketCI v3.7 to 3.8
+
+!!! Important
+    We suggest backing up the KubeRocketCI environment before starting the upgrade procedure.
+
+This section provides detailed instructions for upgrading the EPAM Delivery Platform to version 3.8.1. Follow the steps and requirements outlined below:
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:6]: KubeRocketCI

File: docs/operator-guide/upgrade-edp-3.8.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+  !!! warning
+      This command will delete all Deployments.
+
+    ```bash
+    kubectl -n edp get application -o custom-columns=":metadata.name" | xargs -I {} kubectl -n edp get application {} -o json | jq 'del(.metadata.finalizers)' | kubectl replace -f -
+    ```
+
+  !!! Info
+      Beginning from version 3.8.x KubeRocketCI uses Argo CD application set instead apllications to manage deploy environments, please double check that your upgrade your Argo CD instance to work with this kind of resource. In the [edp-cluster-addons](https://github.com/epam/edp-cluster-add-ons/blob/main/add-ons/argo-cd/values.yaml#L30) repository exists example how to install it.
```
Comment (NikolayMarusenko) [inline docs/operator-guide/upgrade-edp-3.8.md:52]: please provide required version of argocd with application set support

**Review by NikolayMarusenko — APPROVED** [review]

**Review by SergK — APPROVED** [review]

**Pull request merged**